tests/components/eafm/test_sensor.py

```diff
@@ -0,0 +1,319 @@
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+    assert state.attributes["unit_of_measurement"] == "m"
+
+    await poll(
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        }
+    )
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "unavailable"
+
+    await poll(
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        }
+    )
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+
+
+async def test_unload_entry(hass, mock_get_station):
+    """Test being able to unload an entry."""
+    entry, _ = await async_setup_test_fixture(
+        hass,
+        mock_get_station,
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+
+    # Before unloading there should be a station object and it should be polling
+    station = hass.data[DOMAIN]["L1234"]
```

Comment:
We should avoid asserting integration details in tests. The details may change in the future.

Reply:
My motivation at the time was to prove the integration had cleaned up after itself. Wasn't sure how to do this without poking at the internals. Luckily using DataUpdateCoordinator has meant I don't need to do this.